OWML.ModHelper.Menus/ModLayoutButton.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Reflection;
+using OWML.Common.Menus;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.EventSystems;
+
+namespace OWML.ModHelper.Menus
+{
+    public class ModLayoutButton : IModLayoutButton
+    {
+        public event Action OnClick;
+
+        public Button Button { get; }
+        public IModMenu Menu { get; private set; }
+        public HorizontalLayoutGroup LayoutGroup { get; private set; }
+        private UIStyleApplier _buttonStyleApplier;
+
+        private int _index;
+        public int Index
+        {
+            get => Button.transform.parent == null ? _index : Button.transform.GetSiblingIndex();
+            set
+            {
+                _index = value;
+                Button.transform.SetSiblingIndex(value);
+            }
+        }
+        public ModLayoutButton(Button button, IModMenu menu)
+        {
+            Button = button;
+            Button.onClick.AddListener(() => OnClick?.Invoke());
+            var temp = Button.GetComponentInChildren<Text>().gameObject;
+            GameObject.Destroy(temp);
+            temp = new GameObject("LayoutGroup", new Type[] { typeof(RectTransform) });
+            temp.transform.SetParent(button.transform);
+            var img = temp.AddComponent<Image>();
```

Comment:
What is this image for? The name `img` might be a bit too vague. If a descriptive name is not possible, a comment might help.

Reply:
that transparent raycast target, can rename it to target

Reply:
done